common/changes/@cadl-lang/compiler/deprecate-create-program_2022-09-19-15-07.json

```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@cadl-lang/compiler",
+      "comment": "Deprecate `createProgram` and change `compile` parameter order to match `createProgram`",
```

Comment:
I think it's probably most prudent to:

1. Effectively leave compile and createProgram parameter order alone, but...
2. Flip things around so the one with the code is compile and the wrapper is createProgram.
3. Deprecate createProgram wrapper rather than createProgram = compile.

Reply:
it felt like the `createProgram` parameter order was better than `compile` which is why I changed it but yeah does make it weird to have deprecate and breaking at once.

Reply:
So I would say either leave the order or just break it. I don't have a strong opinion. I don't think anyone would be calling createProgram all over the place so maybe the break is OK.

Reply:
It feels like the in between option is like the worst of all worlds.

Reply:
Maybe a (hopefully) very quick design vote on options:

1. PR as-is
2. Deprecate without break
3. Break without deprecate